### PR TITLE
Swagger ui

### LINKED ISSUE #50: Update '/' with documentation info
- [x] Complete '/' page content with endpoint details 
- [x] Look into using Swagger UI

### PR DESCRIPTION
**Pull request by linda-le1**
## What functionality does this accomplish?
closes #50 

**Description:**
Added Swagger UI capabilities found at /api/docs. Reorganized folders into models and services. Add template for homepage to link to Swagger UI. 

## What did you struggle on to complete?
Needed to add CORS (Cross Origin Resource Sharing) to allow Swagger UI to hit the API application. Also had to move static folder into same folder as base_app.py file. 

### Test Coverage Percentage: x%
- [ ] No Tests have been changed
- [ ] New Tests have been implemented
- [X] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened):

## Checklist:
- [ ] My code has no unused/commented out code
- [X] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have fully tested my code
- [ ] I have partially tested my code (please explain why):

## Helpful Resources:
* Resource link AND small description of info gathered/learned
https://swagger.io/docs/open-source-tools/swagger-ui/usage/cors/
https://flask-cors.readthedocs.io/en/latest/
https://exploreflask.com/en/latest/static.html

## Please include a gif of how you feel about this branch:
giphy

### CODE REVIEW

File: base_app/services/yelp_service.py [hunk 1]
```diff
@@ -2,8 +2,8 @@
 import os
 import random
 import requests
-from base_app.restaurant import Restaurant
-from base_app.restaurant_schema import RestaurantSchema
+from base_app.models.restaurant import Restaurant
+from base_app.models.restaurant_schema import RestaurantSchema
```
Comment (freeheeling) [inline base_app/services/yelp_service.py:6]: Makes sense at this stage to break classes out into their respective folder types.

File: base_app/static/swagger.json [hunk 2]
```diff
@@ -0,0 +1,106 @@
+{
+  "swagger" : "2.0",
+  "info" : {
+    "description" : "This is a REST API that consumes the Yelp and Google Map APIs. A restaurant recommendation with details is returned.",
+    "version" : "1.0.0",
+    "title" : "Hangry Ateball",
+    "license" : {
+      "name" : "Apache 2.0",
+      "url" : "http://www.apache.org/licenses/LICENSE-2.0.html"
+    }
```
Comment (freeheeling) [inline base_app/static/swagger.json:10]: Is the Apache license a Swagger requirement?

File: base_app/static/swagger.json [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  },
+  "host" : "hangry-ateball-api.herokuapp.com",
+  "basePath" : "/api/v1",
+  "schemes" : [ "http" ],
+  "paths" : {
```
Comment (freeheeling) [inline base_app/static/swagger.json:15]: We'll need to add the soon to be created `/notify` path to this file as well.

File: base_app/static/swagger.json [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+          "in" : "query",
+          "description" : "Pass an integer (1 - 4) to select a maximum price point ($ - $$$$). For example, when 3 is entered, the search will also include restaurants with a price point of 2 and 1.",
+          "required" : false,
+          "type" : "integer",
+          "minimum" : 1,
+          "maximum" : 4
+        }, {
+          "name" : "travel",
+          "in" : "query",
+          "description" : "Enter either 'walk' or 'drive'. If walk is entered, a radius of 1600 meters will be searched. If drive is entered, a radius of 16,000 meters will be searched.",
```
Comment (freeheeling) [inline base_app/static/swagger.json:84]: I would suggest adding a comma to `1600` for consistency.

File: tests/user_can_see_api_homepage_test.py [hunk 5]
```diff
@@ -12,5 +12,5 @@ def test_homepage_returns_documentation_info():
     url = '/'
     client = app.test_client()
     response = client.get(url)
-    text = response.data
-    assert text == b'<h1>Welcome to the Hangry-8Ball API!</h1><p>Enter API documentation and endpoints here.</p>'
\ No newline at end of file
+    text = response.data.decode('utf8')
```
Comment (freeheeling) [inline tests/user_can_see_api_homepage_test.py:15]: I'm curious now about what the `decode` method does.

**Review by freeheeling — APPROVED** [review]
Great work determining how to implement Swagger UI.

**Review by Jonpatt92 — APPROVED** [review]
This is an excellent refactor and restructuring of our API

**Pull request merged**